prestodb/hdp2.6-hive/Dockerfile

```diff
@@ -0,0 +1,75 @@
+# Copyright 2016 Teradata
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+FROM prestodb/centos6-oj8:unlabelled
+MAINTAINER Presto community <https://prestodb.io/community.html>
+
+# Change default timezone
+RUN ln -snf "/usr/share/zoneinfo/Asia/Kathmandu" /etc/localtime && echo "Asia/Kathmandu" > /etc/timezone
+
+# Install HDP repo
+RUN wget -nv http://public-repo-1.hortonworks.com/HDP/centos6/2.x/updates/2.6.3.0/hdp.repo -P /etc/yum.repos.d
+
+# Install Hadoop, Hive (w/ MySQL)
+RUN yum install -y \
+    hadoop-hdfs-namenode \
+    hadoop-hdfs-secondarynamenode \
+    hadoop-hdfs-datanode \
+
+    hadoop-yarn-resourcemanager \
+    hadoop-yarn-nodemanager \
+
+    hive \
+    hive-metastore \
+    hive-server2 \
+
+    mysql-server mysql-connector-java \
+
+# Cleanup
+  && yum -y clean all && rm -rf /tmp/* /var/tmp/* \
+  && ln -s /usr/share/java/mysql-connector-java.jar /usr/hdp/current/hive-metastore/lib/mysql-connector-java.jar
+
+# Delete original configuration
+RUN rm -r /etc/hadoop/conf/* \
+  && rm -r /etc/hive/conf/*
+
+# Copy configuration files
+COPY ./files /
+
+# Run setup script
+RUN /root/setup.sh \
+  && rm -rf /tmp/* /var/tmp/*
+
+# Setup sock proxy
+RUN yum install -y openssh openssh-clients openssh-server && yum -y clean all
+RUN ssh-keygen -t rsa -b 4096 -C "automation@prestodb.io" -N "" -f /root/.ssh/id_rsa \
+  && ssh-keygen -t rsa -b 4096 -N "" -f /etc/ssh/ssh_host_rsa_key \
+  && ssh-keygen -t dsa -b 1024 -N "" -f /etc/ssh/ssh_host_dsa_key \
+  && cp /root/.ssh/id_rsa.pub /root/.ssh/authorized_keys
+RUN chmod 755 /root && chmod 700 /root/.ssh
+RUN passwd --unlock root
```

Comment:
interesting why it's needed.

also, if this is part sock proxy setup (i don't know, but code layout suggests so),  you can put everything in single RUN

Reply:
I would rather not. It's a compromise between having a huge RUN block, and having many RUNs, I'm trying to group similar commands that may fail because of similar reasons, etc